src/ast.rs

```diff
@@ -325,24 +377,36 @@ impl Expression {
                             .await
                             .map(Expression::Begin)
                     }
-                    [Syntax::Identifier { ident, span, .. }, tail @ .., Syntax::Null { .. }]
+                    [Syntax::Identifier {
+                        ident,
+                        ref mut span,
+                        ..
+                    }, tail @ .., Syntax::Null { .. }]
                         if ident == "lambda" =>
                     {
-                        Lambda::parse(runtime, tail, env, span /* cont */)
+                        Lambda::parse(runtime, tail, env, span.clone() /* cont */)
                             .await
                             .map(Expression::Lambda)
                     }
-                    [Syntax::Identifier { ident, span, .. }, tail @ .., Syntax::Null { .. }]
+                    [Syntax::Identifier {
+                        ident,
+                        ref mut span,
```

Comment:
I think you can remove the `ref mut` here, or at least remove the `mut`.

This appears in a couple of places